API/Services/Repositories/IRepository.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace API.Services.Repositories
+{
+    public interface IRepository<TCollection>
+    {
+        Task<IEnumerable<TCollection>> GetAllAsync();
+        Task<TCollection?> GetAsync(Guid id);
```

Comment:
Also it will be good if we add FindByCondition to get filtered items or we can make it latter And it would be better to use Filter instead of Linq because we use mongodb